src/listpack.c

```diff
@@ -1218,13 +1256,73 @@ int listpackTest(int argc, char *argv[], int accurate) {
     UNUSED(argv);
     UNUSED(accurate);
 
-    TEST("check lpValidateIntegrity") {
-        unsigned char *lp = createList();
+    unsigned char *lp;
+
+    TEST("Test lpFind") {
+        lp = createList();
+        assert(lpFind(lp, lpFirst(lp), (unsigned char*)"abc", 3, 0) == NULL);
+        verifyEntry(lpFind(lp, lpFirst(lp), (unsigned char*)"hello", 5, 0), (unsigned char*)"hello", 5);
+        verifyEntry(lpFind(lp, lpFirst(lp), (unsigned char*)"1024", 4, 0), (unsigned char*)"1024", 4);
+        lpFree(lp);
+    }
+
+    TEST("Test lpValidateIntegrity") {
+        lp = createList();
         long count = 0;
         assert(lpValidateIntegrity(lp, lpBytes(lp), 1, lpValidation, &count) == 1);
         lpFree(lp);
     }
 
+    /* Benchmarks */
+    {
+        int iteration = accurate ? 100000 : 100;
+        lp = lpNew(0);
+        TEST("Benchmark lpPushTail") {
+            unsigned long long start = usec();
+            for (int i=0; i<iteration; i++) {
+                char buf[4096] = "asdf";
+                lp = lpPushTail(lp, (unsigned char*)buf, 4);
+                lp = lpPushTail(lp, (unsigned char*)buf, 40);
+                lp = lpPushTail(lp, (unsigned char*)buf, 400);
+                lp = lpPushTail(lp, (unsigned char*)buf, 4000);
+                lp = lpPushTail(lp, (unsigned char*)"1", 1);
+                lp = lpPushTail(lp, (unsigned char*)"10", 2);
+                lp = lpPushTail(lp, (unsigned char*)"100", 3);
+                lp = lpPushTail(lp, (unsigned char*)"1000", 4);
+                lp = lpPushTail(lp, (unsigned char*)"10000", 5);
+                lp = lpPushTail(lp, (unsigned char*)"100000", 6);
+            }
+            printf("%lld\n", usec()-start);
+        }
+
+        TEST("Benchmark lpFind") {
+            unsigned long long start = usec();
+            for (int i = 0; i < 2000; i++) {
+                unsigned char *fptr = lpFirst(lp);
+                fptr = lpFind(lp, fptr, (unsigned char*)"nothing", 7, 1);
+            }
+            printf("%lld\n", usec()-start);
```

Comment:
considering your optimized lpFind, i think it would be nice to add a benchmark that searches for something numeric too.

Reply:
I have added in redis_test.